components/webgpu/lib.rs

```diff
@@ -1458,3 +1466,33 @@ pub struct PresentationData {
     image_desc: ImageDescriptor,
     image_data: ImageData,
 }
+
+pub trait Transmute<U: id::Marker> {
```

Comment:
Does it not make sense to simply follow the approach of https://github.com/gfx-rs/wgpu/pull/5509/files and use the explicit method calls(such as `into_queue_id`)?

Reply:
Yes, it would, but we are missing `into_device_id`, so we need to roll our own anyway and that way we know that when `into_queue_id` is removed we need to remove queue_id->device_id transmute too.

We should be able to remove all transmute usages anyway: https://github.com/servo/servo/issues/32222.

Reply:
I changed command_buffer<->command_encoder transmutes to into_command_buffer_id, into_command_encoder_id calls.